docs/modules/tls.md

```diff
@@ -318,9 +318,66 @@ at `server-ca.crt` in the root of the nodemcu-firmware build tree. The build scr
 firmware image.
 
 The alternative approach is easier for development, and that is to supply the PEM data as a string value to `tls.cert.verify`. This
-will store the certificate into the flash chip and turn on verification for that certificate. Subsequent boots of the nodemcu can then
+will store the certificate into the flash chip and turn on verification for that certificate. Subsequent boots of the esp can then
 use `tls.cert.verify(true)` and use the stored certificate.
 
+## tls.cert.auth()
+
+Controls the certificate proving process when the esp authenticates against the other side 
+like when establishing mutual authentication as in MQTT.  the tls.cert.auth data is used as the 
+TLS client key and certificate when the NodeMCU device is making a connection to a remote TLS server.
+
```

Comment:
Perhaps just

    Controls the client key and certificate used when the ESP creates a TLS connection (for example,
    through `tls.createConnection` or `https` or `mqtt` connections with `secure = true`).

Reply:
Brilliant!